sycl/include/sycl/info/device_traits.def

```diff
@@ -1,110 +1,119 @@
+__SYCL_PARAM_TRAITS_SPEC(device, partition_type_affinity_domain, info::partition_affinity_domain, PI_DEVICE_INFO_PARTITION_TYPE)
+// Extensions/deprecated
+__SYCL_PARAM_TRAITS_SPEC(device, atomic64, bool, PI_DEVICE_INFO_ATOMIC_64)
+__SYCL_PARAM_TRAITS_SPEC(device, kernel_kernel_pipe_support, bool, 0)
+__SYCL_PARAM_TRAITS_SPEC(device, reference_count, pi_uint32, PI_DEVICE_INFO_REFERENCE_COUNT)
+// To be dropped (has alternatives/not needed)
+__SYCL_PARAM_TRAITS_SPEC(device, usm_device_allocations, bool, PI_USM_DEVICE_SUPPORT)
+__SYCL_PARAM_TRAITS_SPEC(device, usm_host_allocations, bool, PI_USM_HOST_SUPPORT)
+__SYCL_PARAM_TRAITS_SPEC(device, usm_shared_allocations, bool, PI_USM_SINGLE_SHARED_SUPPORT)
+__SYCL_PARAM_TRAITS_SPEC(device, usm_restricted_shared_allocations, bool, PI_USM_CROSS_SHARED_SUPPORT)
+__SYCL_PARAM_TRAITS_SPEC(device, usm_system_allocations, bool, PI_USM_SYSTEM_SHARED_SUPPORT)
+__SYCL_PARAM_TRAITS_SPEC(device, image_max_array_size, size_t, PI_DEVICE_INFO_IMAGE_MAX_ARRAY_SIZE)
+// To be dropped (no alternatives)
+__SYCL_PARAM_TRAITS_SPEC(device, opencl_c_version, std::string, PI_DEVICE_INFO_OPENCL_C_VERSION)
+// Extensions
+__SYCL_PARAM_TRAITS_SPEC(device, sub_group_independent_forward_progress, bool, PI_DEVICE_INFO_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_pci_address, std::string, PI_DEVICE_INFO_PCI_ADDRESS)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_gpu_eu_count, pi_uint32, PI_DEVICE_INFO_GPU_EU_COUNT)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_gpu_eu_simd_width, pi_uint32, PI_DEVICE_INFO_GPU_EU_SIMD_WIDTH)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_gpu_slices, pi_uint32, PI_DEVICE_INFO_GPU_SLICES)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_gpu_subslices_per_slice, pi_uint32, PI_DEVICE_INFO_GPU_SUBSLICES_PER_SLICE)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_gpu_eu_count_per_subslice, pi_uint32, PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_gpu_hw_threads_per_eu, pi_uint32, PI_DEVICE_INFO_GPU_HW_THREADS_PER_EU)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_max_mem_bandwidth, pi_uint64, PI_DEVICE_INFO_MAX_MEM_BANDWIDTH)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_mem_channel, bool, PI_MEM_PROPERTIES_CHANNEL)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_srgb, bool, PI_DEVICE_INFO_IMAGE_SRGB)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_intel_device_info_uuid, detail::uuid_type, PI_DEVICE_INFO_UUID)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_max_global_work_groups, size_t, PI_EXT_ONEAPI_DEVICE_INFO_MAX_GLOBAL_WORK_GROUPS)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_max_work_groups_1d, id<1>, PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_1D)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_max_work_groups_2d, id<2>, PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_2D)
+__SYCL_PARAM_TRAITS_SPEC(device, ext_oneapi_max_work_groups_3d, id<3>, PI_EXT_ONEAPI_DEVICE_INFO_MAX_WORK_GROUPS_3D)
```

Comment:
Not that it should happen in this, but now that we have the option of having templated info descriptors we should consider updating this extension to use that instead of the `_nd` suffix variants.